server/game/cards/01_SOR/events/Restock.ts

```diff
@@ -0,0 +1,39 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { EventCard } from '../../../core/card/EventCard';
+import { RelativePlayer, TargetMode, ZoneName } from '../../../core/Constants';
+
+export default class Restock extends EventCard {
+    protected override getImplementationId() {
+        return {
+            id: '6087834273',
+            internalName: 'restock'
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.setEventAbility({
+            title: 'Choose up to 4 cards in a discard pile. Put them on the bottom of their owner\'s deck in a random order',
+            targetResolver: {
+                mode: TargetMode.Select,
```

Comment:
Eventually I think we will need a selector type that restricts the list of options based on what's already selected. For example, here we could just make all discard cards selectable and then once one is selected it would restrict the choices to be only cards in that same pile. Here it's optional, but I think we'll have to have it for Bold Resistance in set 3.

All that to say, could we add a TODO here for creating a conditional select type

Reply:
Do you think it's a good idea ? That maybe could represent many cards and that could lead to user error if there is no sufficient indication between our card and opponent cards. I think it will be necessary for Bold Resolution but here I don't know when any players will select opponent discard

Reply:
Will continue this discussion in chat to unblock this PR